ravenframework/JobHandler.py

```diff
@@ -162,7 +224,7 @@ def initialize(self):
     # initialize PBS
     with self.__queueLock:
       self.__running       = [None]*self.runInfoDict['batchSize']
-      self.__clientRunning = [None]*self.runInfoDict['batchSize']
+      self.__clientRunning = [None]*self.runInfoDict['batchSize'] * 2
```

Comment:
***WARNING***

@wangcj05 @joshua-cogliati-inl Here it is the ugly thing. I had to add this hardcoded resizing because if we use the LogicalModel/HybridModel withing an ensembleModel the client queue needs to be resized since both models use the Client Queue. If we dont do this, the EnsembleModel submits the job, but the Hybrid/Logical model within it cannot ever be executed since there are no "slots" avaialble for the model to be executed. 

@joshua-cogliati-inl @wangcj05 do you have any idea how we can improve this to make it more maintainable and not so fragile?

Reply:
I am not sure the cleanest way to fix this.  JobHandler has two limits that cause problems.
1. We haven't really made a way to create a second JobHandler, that could then independently run a bunch of clients, and then when finished the second JobHandler can exit and we can get back to the main JobHandler.
2. We don't have a way for something running in a JobHandler slot to "yield". As in if we did we could add a bunch of jobs and then "yield", and then this would free up the slot until the subjobs finished. (This would also need some kind of "join", as in wait until a specific (or several specific) job(s) finish, and then start again.)

So fixing this cleanly might require a bunch of work on the JobHandler.

Reply:
@joshua-cogliati-inl take a look at last commit...it is very ugly/not clean/elegant and it would need quite a bit of work to make it "flawless" but your idea to have "local/second JobHandler" seems to work just fine.

Reply:
Yes, 52306363076e0eeb04b688d652469fb6f32e82a5 looks like the right direction.